packages/plugin-js-packages/src/lib/runner/outdated/transform.ts

```diff
@@ -18,23 +14,27 @@ export function outdatedResultToAuditOutput(
   const relevantDependencies: OutdatedResult = result.filter(
     dep => dep.type === dependencyGroupToLong[depGroup],
   );
-  // TODO use semver logic to compare versions
-  const outdatedDependencies = relevantDependencies
-    .filter(dep => dep.current !== dep.latest)
-    .filter(
-      dep =>
-        dep.current.split('-')[0]?.toString() !==
-        dep.latest.split('-')[0]?.toString(),
-    );
 
-  const outdatedStats = outdatedDependencies.reduce(
-    (acc, dep) => {
-      const outdatedLevel = getOutdatedLevel(dep.current, dep.latest);
-      return { ...acc, [outdatedLevel]: acc[outdatedLevel] + 1 };
-    },
-    { major: 0, minor: 0, patch: 0 },
+  const validDependencies = relevantDependencies
+    .filter(dep => parse(dep.current) != null && parse(dep.latest) != null)
+    .map(dep => ({
+      ...dep,
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      current: clean(dep.current)!,
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      latest: clean(dep.latest)!,
+    }));
+
+  const outdatedDependencies = validDependencies.filter(dep =>
+    neq(dep.current, dep.latest),
   );
 
+  const outdatedStats = outdatedDependencies.reduce((acc, dep) => {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const outdatedLevel = diff(dep.current, dep.latest)!;
+    return { ...acc, [outdatedLevel]: acc[outdatedLevel] + 1 };
+  }, objectFromEntries(RELEASE_TYPES.map(versionType => [versionType, 0])));
```

Comment:
The docs say that `diff()` returns `null` if the versions are the same. That's probably not gonna happen in `npm outdated`'s output. But to be on the safe-side, such `diff()` results can easily be ignored:

```suggestion
  const outdatedStats = outdatedDependencies.reduce((acc, dep) => {
    const outdatedLevel = diff(dep.current, dep.latest);
    if (outdatedLevel == null) {
      return acc;
    }
    return { ...acc, [outdatedLevel]: acc[outdatedLevel] + 1 };
  }, objectFromEntries(RELEASE_TYPES.map(versionType => [versionType, 0])));
```